lib/MonadicRewrite.thy

```diff
@@ -760,6 +760,10 @@ lemma monadic_rewrite_corres_r:
 lemmas corres_gets_the_bind
   = monadic_rewrite_corres_l[OF monadic_rewrite_bind_head[OF monadic_rewrite_gets_the_gets]]
 
+lemma oblivious_monadic_rewrite:
```

Comment:
this is in the corres/refinement section... isn't there a better place for it, even a new section?
btw, this should work for any F, right?

Reply:
It will work with both flags schematic, actually. Thanks.

I could make a new section, but I'm not sure what to call it. Maybe "Interaction with `oblivious`"? Or maybe I could expand the current section to "corres / refinement / oblivious"?

Reply:
I don't think I'd want it in this section, even renamed. My suggestion is to create a section after this one (before Tool Integration), something like: "Interaction with other rewrite techniques: oblivious" (and then if more come along, we can do oblivious/whatever). Thoughts?

Reply:
Sure, I'll go with your suggestion

Reply:
+1 for a new section, I think calling it "Interaction with oblivious" would be fine.

Reply:
Sigh. This is probably the only interaction with oblivious we're going to see, hence my suggestion for a more general section name, but this can work I guess.

Reply:
Happy to call it whatever. If you two could decide between yourselves and let me know, that would be appreciated

Reply:
Gerwin guesses that there might be more rules for oblivious, I'm not so sure. Fine to leave it like it is.